File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/windows/longest_common_subsequence/c#/template.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class Solution {
    public static List<int> longestCommonSubsequence(List<int> a, List<int> b) {
        return new List<int>();
    }

    static void Main(String[] args) {
        string[] tokens = Console.ReadLine()!.Split(' ');
        int n = int.Parse(tokens[0]);
        int m = int.Parse(tokens[1]);

        List<int> a = Console.ReadLine()!.Split(' ').Select(int.Parse).ToList();
        List<int> b = Console.ReadLine()!.Split(' ').Select(int.Parse).ToList();

        List<int> result = longestCommonSubsequence(a, b);

        using (StreamWriter file = new StreamWriter("solution.txt")) {
            file.WriteLine(string.Join(" ", result));
        }
    }
}

File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/windows/longest_common_subsequence/rust/template.rs
use std::io::{self, Read};
use std::fs::File;
use std::io::Write;

fn longest_common_subsequence(a: &[i32], b: &[i32]) -> Vec<i32> {
    vec![]
}

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut tokens = input.split_whitespace();

    if let (Some(n_str), Some(m_str)) = (tokens.next(), tokens.next()) {
        let n: usize = n_str.parse().unwrap();
        let m: usize = m_str.parse().unwrap();

        let mut a = Vec::with_capacity(n);
        for _ in 0..n { a.push(tokens.next().unwrap().parse().unwrap()); }

        let mut b = Vec::with_capacity(m);
        for _ in 0..m { b.push(tokens.next().unwrap().parse().unwrap()); }

        let result = longest_common_subsequence(&a, &b);

        let mut file = File::create("solution.txt").unwrap();
        let strings: Vec<String> = result.iter().map(|n| n.to_string()).collect();
        writeln!(file, "{}", strings.join(" ")).unwrap();
    }
}